---
title: Analysis of epistatic inflation under large single additive effects, with a focus on the replication rate
author: Gibran Hemani
---

```{r}
suppressWarnings(suppressPackageStartupMessages({
	library(knitr)
	library(tidyverse)
}))
opts_chunk$set(cache=TRUE)
```

## Summary

These simulations suggest

- that discovering associations at experiment wide thresholds (e.g. 1e-16) is unlikely even when the lambda is very high
- the replication rate could be quite high using FDR, even if the lambda is very low
- the replication rate is very low at experiment wide thresholds
- we might get higher experiment wide replication rates if we found a way that the discovery effects surpassed the experiment wide threshold


## Background

A 4 d.f. test for epistatic interactions is intended to be orthogonal from additive effects. However we are now seeing that a variant which is in LD with a large causal variant will be susceptible to large 4df test statistics across the genome. This can lead to a high false discovery rate at stringent significance thresholds. If the causal variant is available with low measurement error, fitting it as a covariate is sufficient to eliminate these false positives.

In these simulations we are looking at two things:

1. The influence of the magnitude of the additive effect on the false discovery rate
2. The replication rate, in an independent sample, of false positives from the discovery analysis.

## Simulation scenario

Here we attempt to mimic the scenario in Hemani et al 2014. There were 846 samples in the discovery and a combined 2131 in the replication. We reported the MBNL1 gene being influenced by several cis-trans epistatic interactions, where the cis variant was rs13069559. These associations replicated at the Bonferroni level in the independent replication sample. Wood et al 2014 showed that fitting the fine-mapped additive cis-variant rs67903230 abrogated the cis-trans signals involving rs13069559.

So what we did here was:


1. Use ALSPAC imputed genotype data to create a discovery data set and replication dataset of the same sizes as used in the original analysis. 
2. Simulate a phenotype where the fine-mapped variant, rs67903230, had a large additive effect.
3. Performed a 4df interaction test of the original cis variant, rs13069559, against 502510 genotyped markers in the discovery
4. Tested any variants at Bonferroni significance for replication in the independent dataset

As in the original paper, only interactions between SNP pairs that had all 9 genotype classes were retained, and only one trans variant per chromosome was retained for replication. All SNPs from the cis chromosome except the cis-proxy SNP were excluded, so we're only looking for cis-trans effects here.

We also added an extra variable to test. Most expression traits have a large cis effect and a complex, polygenic trans component. For half of the simulations we also created a polygenic component that explained 30% of the variance, which was composed of 1000 SNPs which each had an effect sampled from a normal distribution.

```{r}
l <- list()
# load("../data/rs67903230.rdata")
# res$polygenic <- res$sim %% 2 == 0
# res$median_p_disc <- pchisq(res$lambda_disc * qchisq(0.5, 1), 1, low=FALSE)
# res$median_p_repl <- pchisq(res$lambda_repl * qchisq(0.5, 1), 1, low=FALSE)
# res$lambda_disc <- pnorm(1 - res$median_p_disc / 2)^2 / qchisq(0.5, 1)
# res$lambda_repl <- pnorm(1 - res$median_p_repl / 2)^2 / qchisq(0.5, 1)
# res$gene <- "MBNL1"
# res$i <- paste(res$gene, res$sentinel, res$cis, sep="_")
# res$scenario <- 1

load("../data/aggregate1.rdata")
res$polygenic <- res$sim %% 2 == 0
res$gene <- sapply(res$i, function(x) strsplit(x, "_")[[1]][1])
res$scenario <- 1
l[[1]] <- res

load("../data/aggregate2.rdata")
res$polygenic <- res$sim %% 2 == 0
res$gene <- sapply(res$i, function(x) strsplit(x, "_")[[1]][1])
res$scenario <- 2
l[[2]] <- res

load("../data/aggregate3.rdata")
res$polygenic <- res$sim %% 2 == 0
res$gene <- sapply(res$i, function(x) strsplit(x, "_")[[1]][1])
res$scenario <- 3
l[[3]] <- res

res <- bind_rows(l)

```

We performed `r nrow(res)` simulations, where the variance explained by the causal variant was sampled uniformly between 0% and 50%

```{r}
ggplot(res, aes(x=varexp)) +
geom_histogram() +
facet_grid(scenario ~ .)
```

The lambdaGC value for the genome wide interaction effects were recorded for each simulation from the discovery and replication datasets


```{r}
ggplot(res, aes(y=lambda_repl, x=lambda_disc)) +
geom_point() +
labs(x = "Genomic inflation (discovery)", y = "Genomic inflation (replication)") +
facet_grid(scenario ~ .)
```

And the lambdaGC value was strongly related to the variance explained by the causal variant

```{r}
ggplot(res, aes(x=varexp, y=lambda_disc)) +
geom_point(size=0.03, alpha=0.1) +
labs(x="Additive variance explained", y="Genomic inflation") +
facet_grid(scenario ~ .) +
theme_bw()
```


The number of significant hits at a Bonferroni threshold (p < 0.05 / 500k):


```{r}
ggplot(res, aes(x=nsig_disc_bonf)) +
geom_histogram(aes(fill=polygenic), position="dodge") +
facet_grid(scenario ~ .)
```

And the number significant at the threshold used in the paper pf p < 1e-16:

```{r}
hist(res$nsig_disc_bonf2, xlim=c(0,max(res$nsig_disc_bonf)))
```

Note that we found no significant hits at an experiment wide threshold, this is following `r nrow(res) * 500000` multiple tests. In the original paper we found 501 hits after performing over `r 1e15` tests, however in these simulations we have looked specifically at interactions between only cis-proxy variants which are in principle vulnerable to cis-trans false positives. The number of tests that we did using cis-proxy variants is likely similar in volume to the number performed empirically (as a small fraction of all SNPs are cis-proxy variants).

There looks to be no affect of having a polygenic trans component, so from now on we will not stratify the results on that basis.

The number of independent discovery associations per simulation was higher with larger causal variant effects (p = `r summary(lm(nsig_disc_bonf ~ varexp, res))$coefficients[2,4]`)

```{r}
res$varexp_bin <- Hmisc::cut2(res$varexp, g=30)
ggplot(res, aes(x=varexp_bin, y=nsig_disc_bonf)) +
geom_boxplot() +
theme(axis.text.x=element_text(angle=90, hjust=0.5, vjust=0.5)) +
labs(x="Additive variance explained", y="Mean number of discovery hits") +
facet_grid(scenario ~ .)
```

The average correlation of F statistics from the discovery and replication analyses (without thresholding) was `r mean(res$fstat_cor, na.rm=TRUE)`

```{r}
hist(res$fstat_cor)
```

```{r}
ggplot(res, aes(x=fstat_cor)) +
geom_histogram(fill="white", colour="black") +
labs(x="Correlation of F statistics between discovery and replication") +
facet_grid(scenario ~ ., scale="free_y")
```

The larger the causal variant effect, the larger the F statistic correlation

```{r}
plot(fstat_cor ~ varexp, res)
```

Here's a look at the average number of discovery hits per simulation, binned by levels of genome wide lambda

```{r}
res$lambda_disc_bin <- Hmisc::cut2(res$lambda_disc, g=30)
res$varexp_bin <- Hmisc::cut2(res$varexp, g=30)
res %>% 
ggplot(aes(x=lambda_disc_bin, y=nsig_disc_bonf)) +
geom_boxplot() +
theme(axis.text.x=element_text(angle=90, hjust=0.5, vjust=0.5)) +
labs(x="Genomic inflation of simulation", y="Mean number of discovery hits") +
facet_grid(scenario ~ .)

res %>% 
ggplot(aes(x=varexp_bin, y=nsig_disc_bonf)) +
geom_boxplot() +
theme(axis.text.x=element_text(angle=90, hjust=0.5, vjust=0.5)) +
labs(x="Variance explained by cis causal variant", y="Mean number of discovery hits") +
facet_grid(scenario ~ .)
```


## Is the replication rate high under the null simulations?

If the false positive rate is such that we obtain at least one false positive cis-trans association in the discovery, what is the chance of it replicating in an independent dataset?

We can attempt to replicate at

* Bonferroni threshold (within one simulation, how many discovery associations were replicated at Bonferroni corrected p < 0.05)
* FDR (within one simulation, how many discovery associations replicate at FDR < 0.05).
* experiment threshold (within one simulation, how many discovery associations replicated at p < 0.05/501, the threshold used in the paper)


```{r, fig.width=12}
res$bonf_bonf_rr <- res$cond_bonf_bonf / res$nsig_disc_bonf
res$bonf_fdr_rr <- res$cond_bonf_fdr / res$nsig_disc_bonf
res$bonf_bonf2_rr <- res$cond_bonf_bonf2 / res$nsig_disc_bonf

repl_rate <- bind_rows(
	group_by(res, varexp_bin, scenario) %>% summarise(m=mean(bonf_bonf_rr, na.rm=T)) %>% mutate(threshold="Bonferroni"),
	group_by(res, varexp_bin, scenario) %>% summarise(m=mean(bonf_fdr_rr, na.rm=T)) %>% mutate(threshold="FDR"),
	group_by(res, varexp_bin, scenario) %>% summarise(m=mean(bonf_bonf2_rr, na.rm=T)) %>% mutate(threshold="Experiment")
)
repl_rate %>% ggplot(aes(x=varexp_bin, y=m, group=threshold)) +
geom_line(aes(colour=threshold)) +
scale_colour_brewer(type="qual") +
theme(axis.text.x=element_text(angle=90, hjust=0.5, vjust=0.5)) +
labs(x="Variance explained by cis additive variant", y="Replication rate") +
facet_grid(. ~ scenario)
```

Overall, using FDR there is a reasonably high replication rate which doesn't correspond to discovery lambda. But the experiment wide threshold can only be met when the discovery lambda is very high, but even then the replication rate is extremely low.


## Replication rate of MBNL1

Analysing the MBNL1 locus - what is the discovery and replication rate?

There are 40000 simulations, where the cis-additive variant explains some amount of the variance in the phenotype (ranging from 0% to 50%). If a signal is found to be significant then it is tested for replication. In this plot, the x axis shows how many independent associations found to be significant for a simulations, and the y axis is the proportion of those that replicate at p < 0.05/501. The simulations are binned based on the simulated additive variance (columns of plots). The blue dots are from the simulations, and the red dot is the empirical result for MBNL1. Note that the cis-additive variant explains 10.5% of the variance of MBNL1, so the relevent boxes for comparison against the simulations are the second and third from the left. There are very few discovered loci here, and the replication rate of those is always zero. When the additive locus explains more variance, then the number of discovered hits increases and the replication rate increases.

```{r, fig.width=12}
temp <- subset(res, scenario == 1)

temp$vbin <- cut(temp$varexp, breaks=10)
o <- group_by(temp, vbin, cond_bonf_bonf2, nsig_disc_bonf) %>%
summarise(n=n())
o$what <- "simulation"

e <- tibble(cond_bonf_bonf2=5, nsig_disc_bonf=11, n=1, what="empirical", vbin = unique(temp$vbin))
oe <- bind_rows(o,e)
oe$rr <- oe$cond_bonf_bonf2 / oe$nsig_disc_bonf
oe$rr[is.na(oe$rr)] <- 0

ggplot(oe, aes(x=nsig_disc_bonf, y=rr)) +
geom_point(aes(color=what, size=n)) +
labs(x="Number discovered", y="Proportion replicated") +
facet_grid(. ~ vbin)
```

