---
title: Analysis of epistatic inflation under large single additive effects, with a focus on the replication rate
author: Gibran Hemani
---

## Background

A 4 d.f. test for epistatic interactions is intended to be orthogonal from additive effects. However we are now seeing that a variant which is in LD with a large causal variant will be susceptible to large 4df test statistics across the genome. This can lead to a high false discovery rate at stringent significance thresholds. If the causal variant is available with low measurement error, fitting it as a covariate is sufficient to eliminate these false positives.

In these simulations we are looking at two things:

1. The influence of the magnitude of the additive effect on the false discovery rate
2. The replication rate, in an independent sample, of false positives from the discovery analysis.

## Simulation scenario

Here we attempt to mimic the scenario in Hemani et al 2014. There were 846 samples in the discovery and a combined 2131 in the replication. We reported the MBNL1 gene being influenced by several cis-trans epistatic interactions, where the cis variant was rs13069559. These associations replicated at the Bonferroni level in the independent replication sample. Wood et al 2014 showed that fitting the fine-mapped additive cis-variant rs67903230 abrogated the cis-trans signals involving rs13069559.

So what we did here was:


1. Use ALSPAC imputed genotype data to create a discovery data set and replication dataset of the same sizes as used in the original analysis. 
2. Simulate a phenotype where the fine-mapped variant, rs67903230, had a large additive effect.
3. Performed a 4df interaction test of the original cis variant, rs13069559, against 502510 genotyped markers in the discovery
4. Tested any variants at Bonferroni significance for replication in the independent dataset

As in the original paper, only interactions between SNP pairs that had all 9 genotype classes were retained, and only one trans variant per chromosome was retained for replication. All SNPs from the cis chromosome except the cis-proxy SNP were excluded, so we're only looking for cis-trans effects here.

```{r}
library(tidyverse)
load("../data/rs67903230.rdata")
```

We performed `r nrow(res)` simulations, where the variance explained by the causal variant was sampled uniformly between 0% and 50%

```{r}
hist(res$varexp)
```

The lambdaGC value for the genome wide interaction effects were recorded for each simulation from the discovery and replication datasets


```{r}
plot(lambda_disc ~ lambda_repl, res)
```

And the lambdaGC value was strongly related to the variance explained by the causal variant

```{r}
plot(lambda_disc ~ varexp, res)
```

The number of significant hits at a Bonferroni threshold (p < 0.05 / 500k):


```{r}
hist(res$nsig_disc_bonf)
```

And the number significant at the threshold used in the paper pf p < 1e-16:

```{r}
hist(res$nsig_disc_bonf2, xlim=c(0,max(res$nsig_disc_bonf)))
```

Note that we found no significant hits at an experiment wide threshold, this is following `r nrow(res) * 500000` multiple tests. In the original paper we found 501 hits after performing over `r 1e15` tests, however in these simulations we have looked specifically at interactions between only cis-proxy variants which are in principle vulnerable to cis-trans false positives. The number of tests that we did using cis-proxy variants is likely similar in volume to the number performed empirically (as a small fraction of all SNPs are cis-proxy variants).


The number of independent discovery associations per simulation was higher with larger causal variant effects (p = `r summary(lm(nsig_disc_bonf ~ varexp, res))$coefficients[2,4]`)

```{r}
plot(nsig_disc_bonf ~ varexp, res)
```

The average correlation of F statistics from the discovery and replication analyses (without thresholding) was `r mean(res$fstat_cor, na.rm=TRUE)`

```{r}
hist(res$fstat_cor)
```

The larger the causal variant effect, the larger the F statistic correlation

```{r}
plot(fstat_cor ~ varexp, res)
```

## Is the replication rate high under the null simulations?

If the false positive rate is such that we obtain at least one false positive cis-trans association in the discovery, what is the chance of it replicating in an independent dataset?

We can attempt to replicate at a Bonferroni threshold (within one simulation, how many associations were attempted to replicate at corrected p < 0.05) or FDR (within one simulation, how many associations repliace at FDR < 0.05).

Note that these are much more relaxed thresholds than what we used in the original paper - which was 0.05 / 501.


```{r}

res$lambda_disc_bin <- cut(res$lambda_disc, 5)
temp <- res %>%
	group_by(lambda_disc_bin, nsig_disc_bonf) %>%
	summarise(count=n(), rate=n()/sum(res$lambda_disc_bin == .$lambda_disc_bin[1]))

ggplot(temp %>% filter(nsig_disc_bonf > 0), aes(x=lambda_disc_bin, y=count)) +
geom_bar(stat="identity", aes(fill=nsig_disc_bonf))

table(res$lambda_disc_bin)
hist(res$lambda_disc)

plot(cond_bonf_bonf2 ~ varexp, res)

plot(I(cond_bonf_bonf2/nsig_disc_bonf) ~ varexp, res)

mean(res$cond_bonf_bonf / res$nsig_disc_bonf, na.rm=T)
mean(res$cond_bonf_bonf2 / res$nsig_disc_bonf, na.rm=T)

res$bonf_bonf_rr <- res$cond_bonf_bonf / res$nsig_disc_bonf
res$bonf_fdr_rr <- res$cond_bonf_fdr / res$nsig_disc_bonf
res$bonf_bonf2_rr <- res$cond_bonf_bonf2 / res$nsig_disc_bonf



mean(res$bonf_bonf_rr, na.rm=TRUE)
mean(res$bonf_bonf2_rr, na.rm=TRUE)
tapply(res$bonf_bonf_rr, res$lambda_disc_bin, function(x) mean(x, na.rm=T))
tapply(res$bonf_fdr_rr, res$lambda_disc_bin, function(x) mean(x, na.rm=T))

tapply(res$bonf_bonf_rr, res$nsig_disc_bonf, function(x) mean(x, na.rm=T))
tapply(res$bonf_fdr_rr, res$nsig_disc_bonf, function(x) mean(x, na.rm=T))



```