\documentclass{article}
\usepackage[superscript,biblabel]{cite}
\bibliographystyle{naturemag}

\title{Online methods}
\date{}
\author{}

\begin{document}
\maketitle


\section{Discovery data}

The Brisbane Systems Genetics Study (BSGS) comprises 846 individuals of European descent from 274 independent families \cite{Powell2012}. DNA samples from each individual were genotyped on the Illumina 610-Quad Beadchip by the Scientific Services Division at deCODE Genetics Iceland. Full details of genotyping procedures are given in Medland et al. \cite{Medland2009} Standard quality control (QC) filters were applied and the remaining 528,509 autosomal SNPs were carried forward for further analysis. 

Gene expression profiles were generated from whole blood collected with PAXgene TM tubes (QIAGEN, Valencia, CA) using Illumina HT12-v4.0 bead arrays. The Illumina HT-12 v4.0 chip contains 47,323 probes, although some probes are not assigned to RefSeq genes. We removed any probes that did not match the following criteria: contained a SNP within the probe sequence with MAF $ > 0.05$ within 1000 genomes data; did not map to a listed RefSeq gene; were not significantly expressed (based on a detection $p$-value $< 0.05$) in at least 90\% of samples. After this stringent QC 7339 probes remained for 2D-eQTL mapping.


\section{Normalisation}

Gene expression profiles were normalised and adjusted for batch and polygenic effects. Profiles were first adjusted for raw background expression in each sample. Expression levels were then adjusted using quantile and log2 transformation to standardise distributions between samples. Batch and polygenic effects were adjusted using the linear model

\begin{equation}
y = \mu + \beta_{1}c + \beta_{2}p + \beta_{3}s + \beta_{4}a + g + e
\label{eq:lm}
\end{equation}
where $\mu$ is the population mean expression levels, $c$, $p$, $s$ and $a$ are vectors of chip, chip position, sex and generation respectively, fitted as fixed effects; and $g$ is a random additive polygenic effect with a variance covariance matrix 
\begin{equation}
G_{ijk} = \left \{ 
\begin{array}{ll}
\sigma _a ^2 + \sigma _e ^2&        j = k \\ 
2\phi _{ijk} \sigma _a ^2& 			j \neq k \\
\end{array} \right.
\end{equation}
The parameters $\sigma_a^2$ and $\sigma_e^2$ are variance components for additive background genetic and environmental effects respectively. Here, we are using family based pedigree information rather than SNP based IDB to account for relationships between individuals and so $\phi _{ijk}$ is the kinship coefficient between individuals $j$ and $k$. The residual, $e$, from equation \ref{eq:lm} is assumed to follow a multivariate normal distribution with a mean of zero. Residuals were normalised by rank transformation and used as the adjusted phenotype for the pairwise epistasis scan. The GenABEL package for R was used to perform the normalisation \cite{Aulchenko2007}.


\section{Exhaustive 2D-eQTL analysis}

We used epiGPU \cite{Hemani2011} software to perform an exhaustive scan for pairwise interactions, such that each SNP is tested against all other SNPs for statistical association with the expression values for each of the 7339 probes. For each SNP pair there are 9 possible genotype classes. We treat each genotype class as a fixed effect and fit an 8 \emph{d.f.} $F$-test to test the following hypotheses:

\begin{equation}
H _0 : \sum _{i=1} ^3 \sum _{j=1} ^3 (\bar x _{ij} - \mu) ^2 = 0; 
\end{equation}

\begin{equation}
H _1 : \sum _{i=1} ^3 \sum _{j=1} ^3 (\bar x _{ij} - \mu) ^2 > 0; 
\label{eq:8df}
\end{equation}

where $\mu$ is the mean expression level and $x _{ij}$ is the pairwise genotype class mean for genotype $i$ at SNP 1 and genotype $j$ at SNP 2. This type of test does not parameterize for specific types of epistasis, rather it tests for the joint genetic effects at two loci. This has been demonstrated to be statistically more efficient when searching for a wide range of epistatic patterns, although will also include any marginal effects of SNPs which must be dealt with post-hoc \cite{Hemani2013}.
 
The complete exhaustive scan for 7339 probes comprises $1.03 \times 10^{15}$ $F$-tests. We used permutation analysis to estimate an appropriate significance threshold for the study. To do this we performed a further 1600 exhaustive 2D scans on permuted phenotypes to generate a null distribution of the extreme $p$-values expected to be obtained from this number of multiple tests given the correlation structure between the SNPs. We took the most extreme $p$-value from each of the 1600 scans and set the 5\% FWER to be the 95\% most extreme of these $p$-values, $T_{*} = 2.13 \times 10^{-12}$. The effective number of tests in one 2D scan being performed is therefore $N_{*} = 0.05 / T_{*} \approx 2.33 \times 10^{10}$. To correct for the testing of multiple traits we established an experiment wide threshold of $T = 0.05 / (N_{*} \times 7339) = 2.91 \times 10^{-16}$. This is likely to be conservative as it assumes independence between probes.

Following filtering on this threshold only SNP pairs with at least 5 data points in all 9 genotype classes were kept. We then calculated the LD between interacting SNPs and removed any pairs with $r^2 > 0.1$. If multiple SNP pairs were present on the same chromosomes for a particular expression trait then only the sentinal SNP pair was retained, \emph{i.e.} if a probe had multiple SNP pairs that were on chromosomes one and two then only the SNP pair with the most significant $p$-value was retained. At this stage 6404 filtered SNP pairs remained. We then performed a second filtering screen that was identical to the first but an additional step was included where any SNPs that had previously been shown to have a significant additive or dominant effect ($p < 1.29 \times 10^{-11}$) was removed, creating a second set of 4751 filtered SNP pairs.

To ensure that interacting SNPs were driven by epistasis and not marginal effects we performed a nested ANOVA on each pair in the filtered set to test if the interaction terms were significant. We did this by contrasting the full genetic model (8 \emph{d.f.}) against the reduced marginal effects model which included the additive and dominance terms at both SNPs (4 \emph{d.f.}). Thus, a 4 \emph{d.f.} $F$-test was performed on the residual genetic variation, representing the contribution of epistatic variance. Significance of epistasis was determined using a Bonferroni threshold of $0.05 / (6404+4751) = 4.48 \times 10^{-6}$. This resulted in 432 and 117 SNP pairs with significant interaction terms from filters 1 and 2, respectively.

All variance decompositions were calculated using the NOIA method \cite{Alvarez-Castro2008}.


\section{Replication}
We have attempted replication of the 529 significant interactions from the discovery set using three independent cohorts; Fehrmann, EGCUT, and CHDWB. It was required that LD $r^2 < 0.1$ between interacting SNPs, and all nine genotype classes had at least 5 individuals present in order to proceed with statistical testing for replication in both datasets. We also excluded any putative SNPs that had discordant allele frequencies in any of the datasets. Details of the cohorts are as follows.

\paragraph{Fehrmann: $n=1240$}
The Fehrmann dataset \cite{Fehrmann2011} consists of whole peripheral blood samples of 1240 unrelated individuals from the United Kingdom and the Netherlands. Some of these individuals are patients, while others are healthy controls. Individuals were genotyped using the Illumina 610 Quad platform. RNA levels were quantified using the HT12v3 platform.

\paragraph{EGCUT: $n=891$}
The Estonian Genome Center of the University of Tartu (EGCUT) study \cite{Metspalu2004} consists of whole blood samples of 891 unrelated individuals from Estonia. They were genotyped using the Illumina 610 Quad platform. RNA levels were quantified using the HT12v3 platform.

\paragraph{CDHWB: $n=139$}
The Center for Health Discovery and Well Being (CDHWB) Study \cite{Preininger2013} is a population based cohort consisting of 139 individuals of European descent collected in Atlanta USA. Gene expression profiles were generated with Illumina HT-12 V3.0 arrays from whole blood collected from Tempus tubes that preserve RNA. Whole genome genotypes were measured using Illumina OmniQuad arrays. Due to the small sample size, most SNP pairs did not pass filtering in this dataset and so we have excluded it from the rest of the analysis.

\subsection{Meta Analysis}
The 4 \emph{d.f.} interaction $p$-values for each independent replication dataset were calculated using the same statistical test as was performed in the discovery dataset. We then took the interaction $p$-values from EGCUT and Fehrmann and calculated a joint $p$-value as $-2 \log p_{1} - 2 \log p_{2} \sim \chi^{2}_{4 d.f.}$.


\section{Additive vs epistatic variance comparison}

To compare the relative contribution to the phenotypic variance of gene expression levels between additive and epistatic effects we are constrained by the problem that non-additive variance components for a phenotype cannot be calculated directly. Here, we only have SNP pairs that exceed a threshold of $p < 2.91 \times 10^{-16} = T_{e}$. A strong conclusion cannot be made about the genome-wide variance contribution, but we can compare the variance explained by SNP effects at this threshold for additive scans and epistatic scans. 

In Powell \emph{et al} 2012 \cite{Powell2012} an expression quantitative trait locus (eQTL) study was performed searching for additive effects in the same BSGS dataset as was used for the discovery here. Using the threshold $T_{e}$ for the additive eQTL study, 453 of the 7339 probes analysed here had at least one significant additive effect. Assuming that the phenotypic variance for each of the probes is normalised to 1, the total phenotypic variance of all 7339 explained by the significant additive effects was 1.73\%.

Following the same procedure, at the threshold $T_{e}$ there were 238 gene expression probes with at least one significant pairwise epistatic interaction out of the 7339 tested. In total the proportion of the phenotypic variance explained by the epistatic effects at these SNP pairs was 0.25\%.


\section{Functional enrichment analysis}

\subsection{Tissue specific transcriptionally active regions}

We employed a recently published method (http://www.broadinstitute.org/mpg/epigwas/) \cite{Trynka2013} that tests for cell-type-specific enrichment of active chromatin, measured through H3K4me3 chromatin marks \cite{Koch2007} in regions surrounding the 731 SNPs that comprise the 501 discovery interactions. We subsequently refer to these 731 SNPs as the IndexSNPs A recent paper shows that H3K4me3 chromatin marks are the most cell-type-specific marks in terms of co-localization with previously published GWAS loci \cite{Trynka2013}. The idea behind the method is that variants related to a particular phenotype may affect cell-type-specific gene expression by changing regulatory elements in cell types relevant to that phenotype. Hence, overlap between associated variants and chromatin marks should occur preferentially in the relevant cell type(s). Our analysis tested for enrichment of these chromatin marks in 34 different cell types from 4 different broad tissue types.

For each IndexSNP, a locus region was defined, bounded on either side of the IndexSNP by the most distant SNP within 250 kb that was in LD (r2 > 0.8). For each SNP in a locus region regulatory activity scores were calculated as the height of the nearest H3K4me3 mark divided by distance from the SNP to the H3K4me3 mark. The SNP with the highest score within each locus region in a given tissue was designated the BestSNP, which served as the score representing that locus.

Cell-type-specificity scores per locus were estimated by normalizing BestSNP scores so that the sum of scores for a given locus across all cell types equaled 1. Cell-type-specificity scores per tissue were defined by summing normalized BestSNP scores across all loci within a given tissue. 10,000 sets of SNPs (matched to the IndexSNP regions having the same total number of H3K4me3 peaks) were sampled (from among background SNPs provided with the software) to estimate null distributions of cell-type-specificity scores per locus and per tissue. P-values for cell-type-specificity scores summed across all BestSNPs for each tissue (the observed per-tissue score) were estimated as the proportion of random SNP sets with a per-tissue score exceeding the observed per- tissue score. We identified loci with BestSNP cell-type-specificity scores falling at or above the 95th percentile of the corresponding null distribution as demonstrating greater than expected specificity within a given cell type (50).
Figure S20 shows p-values for the cell-type-specific overlap of H3K4me3 marks and IndexSNP regions for each cell type. Four cell/tissue types showed significant overlap at nominal p â‰¤ 0.05: for EduYears, anterior caudate (p = 0.0089), CD4+ naive primary cells (p = 0.032), hippocampus middle (p = 0.05) and muscle satellite cultured cells (p = 0.0236); and for College, anterior caudate (p = 0.0007). Additionally, for College the mid-frontal lobe showed marginal enrichment at p = 0.0502. Only the anterior caudate results for College survive correction for multiple hypothesis testing.
The results from the analysis of overlap between H3K4me4 chromatin marks and education-associated SNP regions suggest gene expression regulatory function for those loci in specific cell types. In particular, we note that anterior caudate tissue appears enriched for both EduYears and College phenotypes, although only the latter survives multiple testing correction. Figure S21 shows cell-type-specificity scores per locus in the four nominally significant tissues and 95th-percentile threshold (dashed red line). Loci above the threshold were identified as showing greater than chance specificity within that particular cell type. Table S19 identifies these enriched loci, along with distance to the nearest chromatin mark.


\subsection{Chromosome interactions}

I like when chromosomes interact because I like epistasis. Also there is a cool figure with lots of circles.

\subsection{Transcription factor enrichment}

\subsection{SNP colocalisation with genic features}





\section{References}
\bibliography{refs}


\end{document}